### DISCUSSION TITLE
ILLA 本地开发问题

### DISCUSSION THREAD

**Discussion by PeanutQAQ** [illacloud/illa-builder#704] [Q&A] [answered]
1. 问题描述
我在本地 clone 了整个仓库，pnpm install 后，pnpm run dev。
此时本地页面为 https://localhost/api/v1/auth/signin 登录界面，我尝试使用我的账号登录，但是报 net::ERR_CONNECTION_REFUSED 网络异常，请问如何才能启动本地的开发环境那。

**Comment by AruSeito** [comment, marked as answer]
您好，在本地开发时需要您自行配置服务端 API 地址。
你可以在 apps/builder 下建立文件 `.env.cloud.local`
```
VITE_API_BASE_URL=localhost/api/v1
VITE_PROXY_API_BASE_URL=xxxxx.com
```
目前不建议您进行本地开发，本地开发流程配置比较繁琐，后续才会对这个进行优化。建议您尝试使用 illa-cli 进行部署。

**Reply by timeTravelCYN** [reply]
> 您好，在本地开发时需要您自行配置服务端 API 地址。 你可以在 apps/builder 下建立文件 `.env.cloud.local`
> 
> ```
> VITE_API_BASE_URL=localhost/api/v1
> VITE_PROXY_API_BASE_URL=xxxxx.com
> ```
> 
> 目前不建议您进行本地开发，本地开发流程配置比较繁琐，后续才会对这个进行优化。建议您尝试使用 illa-cli 进行部署。

那请问咱们目前可以做到，illa-cli 启动服务，本地开发吗～我想尝试一下看看有没有能做的，但是我现在也卡在了这个环节。顺着 discussion 找到了这里。 ps：目前还没有使用 illa-cli，但我理解那个是部署，是直接跑成品了

**Reply by AruSeito** [reply]
> > 您好，在本地开发时需要您自行配置服务端 API 地址。 你可以在 apps/builder 下建立文件 `.env.cloud.local`
> > ```
> > VITE_API_BASE_URL=localhost/api/v1
> > VITE_PROXY_API_BASE_URL=xxxxx.com
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 目前不建议您进行本地开发，本地开发流程配置比较繁琐，后续才会对这个进行优化。建议您尝试使用 illa-cli 进行部署。
> 
> 那请问咱们目前可以做到，illa-cli 启动服务，本地开发吗～我想尝试一下看看有没有能做的，但是我现在也卡在了这个环节。顺着 discussion 找到了这里。 ps：目前还没有使用 illa-cli，但我理解那个是部署，是直接跑成品了


您好，目前这些工作都还没做，不过如果您想要本地启动项目的话您可以在 clone 该项目后按照如下流程

```
git submodule init && git submodule update
pnpm install
```

然后新建`.env.development.local`文件

```
VITE_API_BASE_URL=localhost:3000/api/v1
VITE_INSTANCE_ID=CLOUD
VITE_PROXY_API_BASE_URL=本地用 cli 启动的后端得到的局域网地址
```

最终执行`pnpm dev`
您可以尝试这样来做一下。
